File: src/main/java/destiny/astrology/eclipse/LunarEclipsePartialObservation.java
/**
 * Created by smallufo on 2017-11-18.
 */
package destiny.astrology.eclipse;

import destiny.astrology.Azimuth;

public class LunarEclipsePartialObservation extends LunarEclipsePenumbraObservation {

  private final boolean partialBeginVisible;

  private final boolean partialEndVisible;

  public LunarEclipsePartialObservation(double gmtJulDay, double lng, double lat, double alt, Azimuth azimuth, AbstractLunarEclipse.LunarType lunarType, double magUmbra, double magPenumbra, boolean maxVisible, boolean penumbraBeginVisible, boolean penumbraEndVisible, boolean partialBeginVisible, boolean partialEndVisible) {
    super(gmtJulDay, lng, lat, alt, azimuth, lunarType, magUmbra, magPenumbra, maxVisible, penumbraBeginVisible, penumbraEndVisible);
    this.partialBeginVisible = partialBeginVisible;
    this.partialEndVisible = partialEndVisible;
  }

  public boolean isPartialBeginVisible() {
    return partialBeginVisible;
  }

  public boolean isPartialEndVisible() {
    return partialEndVisible;
  }
}

File: src/main/java/destiny/astrology/eclipse/LunarEclipsePartialObservation.kt
/**
 * Created by smallufo on 2017-11-18.
 */
package destiny.astrology.eclipse

import destiny.astrology.Azimuth

/** 月偏食 */
open class LunarEclipsePartialObservation(gmtJulDay: Double, lng: Double, lat: Double, alt: Double, azimuth: Azimuth,
                                          lunarType: ILunarEclipse.LunarType,
                                          magUmbra: Double,
                                          magPenumbra: Double,
                                          maxVisible: Boolean,
                                          penumbraBeginVisible: Boolean,
                                          penumbraEndVisible: Boolean,
                                          val isPartialBeginVisible: Boolean,
                                          val isPartialEndVisible: Boolean) :
  LunarEclipsePenumbraObservation(gmtJulDay, lng, lat, alt, azimuth, lunarType, magUmbra, magPenumbra, maxVisible, penumbraBeginVisible, penumbraEndVisible)
